### ISSUE TITLE
Steps not being executed when running concurrent workflows

### ISSUE THREAD

**Issue by karl-diggle** [danielgerlag/workflow-core#536]
We are using workflow to orchestrate RabbitMq messages, and I am currently experiencing steps not being executed intermittently when running concurrent workflows in a Docker container.  

I have used logs to determine that the message received from rabbit was successful and published an event onto the instance of the workflow, but the next expected step is not executed in some instances.

I am triggering 20 workflows concurrently using 2 different definitions and workflow is only completing partially for some instances.  However, when running workflow outside of docker workflow completes successfully every time.

I can confirm that no errors are occurring in any of the Steps as I have added a global error handler and also no errors are being persisted in the database (PostgreSQL persistence).  Just to confirm the data in the persistence does not show the next step in flow.

I am currently running using single-node and therefore using the defaults for everything else.

**Comment by send2vinnie** [comment]
Hi, how do you have your workflows configured? Do you use proper queue provider and distributed lock manager?

https://workflow-core.readthedocs.io/en/latest/multi-node-clusters/

**Comment by karl-diggle** [comment]
Yes I tested it with in-memory initially then I used rabbitmq as the queueprovider and redis for the lock manager and experienced the same issues in both
